opengrok-indexer/src/main/java/org/opengrok/indexer/history/HistoryGuru.java

```diff
@@ -480,7 +478,25 @@ public boolean hasAnnotation(File file) {
      * @return collection of added repositories
      */
     public Collection<RepositoryInfo> addRepositories(File[] files) {
-        return addRepositories(files, env.getNestingMaximum(), 0, false);
+        ExecutorService executor = env.getIndexerParallelizer().getRepositorySearchExecutor();
```

Comment:
I was suggesting just to use `getFixedExecutor()` since it's fully available at this point during a run. It would be fine to leave it to be used later during indexing. `IndexerParallelizer` could be kept simple, and no additional configuration would be needed.

Reply:
I see, makes sense.